From 4859da1da371d75cea2caccc632de17ffc39dc99 Mon Sep 17 00:00:00 2001
From: Wright Feng <wright.feng@cypress.com>
Date: Thu, 26 Apr 2018 01:35:53 -0500
Subject: [PATCH 4696/5242] MLK-19517-03 revert: brcmfmac: add a module
 parameter to set scheduling priority of sdio_dpc

commit  576420bd58c3cf84ecc5f590775b57a511e4216c from
https://source.codeaurora.org/external/imx/linux-imx.git

will use WQ_HIGHPRI instead.

Signed-off-by: Wright Feng <wright.feng@cypress.com>
Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../wireless/broadcom/brcm80211/brcmfmac/common.c  |    5 -----
 .../wireless/broadcom/brcm80211/brcmfmac/sdio.c    |   10 ----------
 2 files changed, 15 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
index 9d89886..a12945a 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/common.c
@@ -82,10 +82,6 @@
 module_param_named(eap_restrict, brcmf_eap_restrict, int, 0400);
 MODULE_PARM_DESC(eap_restrict, "Block non-802.1X frames until auth finished");
 
-static int brcmf_sdio_dpc_prio;
-module_param_named(sdio_dpc_prio, brcmf_sdio_dpc_prio, int, S_IRUSR);
-MODULE_PARM_DESC(sdio_dpc_prio, "The scheduling priority of sdio_dpc thread");
-
 #ifdef DEBUG
 /* always succeed brcmf_bus_started() */
 static int brcmf_ignore_probe_fail;
@@ -437,7 +433,6 @@ struct brcmf_mp_device *brcmf_get_module_param(struct device *dev,
 	settings->roamoff = !!brcmf_roamoff;
 	settings->eap_restrict = !!brcmf_eap_restrict;
 	settings->iapp = !!brcmf_iapp_enable;
-	settings->sdio_dpc_prio = brcmf_sdio_dpc_prio;
 #ifdef DEBUG
 	settings->ignore_probe_fail = !!brcmf_ignore_probe_fail;
 #endif
diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
index 8b08e14..6525286 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/sdio.c
@@ -33,7 +33,6 @@
 #include <linux/bcma/bcma.h>
 #include <linux/debugfs.h>
 #include <linux/vmalloc.h>
-#include <uapi/linux/sched/types.h>
 #include <asm/unaligned.h>
 #include <defs.h>
 #include <brcmu_wifi.h>
@@ -2696,18 +2695,9 @@ static void brcmf_sdio_dpc(struct brcmf_sdio *bus)
 	uint txlimit = bus->txbound;	/* Tx frames to send before resched */
 	uint framecnt;			/* Temporary counter of tx/rx frames */
 	int err = 0;
-	struct sched_param param;
-	struct brcmf_pub *drvr = bus->sdiodev->bus_if->drvr;
-	int sdio_dpc_prio = drvr->settings->sdio_dpc_prio;
 
 	brcmf_dbg(SDIO, "Enter\n");
 
-	if (sdio_dpc_prio > 0) {
-		param.sched_priority = (sdio_dpc_prio < MAX_RT_PRIO) ?
-				  sdio_dpc_prio : (MAX_RT_PRIO - 1);
-		sched_setscheduler(current, SCHED_FIFO, &param);
-	}
-
 	sdio_claim_host(bus->sdiodev->func1);
 
 	/* If waiting for HTAVAIL, check status */
-- 
1.7.9.5

